printing/printer.ml

```diff
@@ -494,15 +494,18 @@ let pr_transparent_state (ids, csts) =
 	str"CONSTANTS: " ++ pr_cpred csts ++ fnl ())
 
 (* display complete goal *)
-let default_pr_goal gs =
+let pr_goal ?(diffs=None) gs =
   let g = sig_it gs in
   let sigma = project gs in
   let env = Goal.V82.env sigma g in
   let concl = Goal.V82.concl sigma g in
-  let goal =
-    pr_context_of env sigma ++ cut () ++
-      str "============================" ++ cut ()  ++
-      pr_goal_concl_style_env env sigma concl in
+  let goal = begin match diffs with
+    | Some pp -> pp
+    | None ->
```

Comment:
This doesn't seem to make sense, you pass an optional parameter that if set will ignore the function completely, yet you still do work before?

Revert and let's the caller choose whether to call the function.

Reply:
For this one the word "revert" makes sense (see comment below).